### PR TITLE
Samples: Bluetooth: Mesh: Use overlay for DFU in Light sample

### PR DESCRIPTION
**Pull request by ludvigsj**
This makes the DFU-specific project configuration an extra configuration file overlay instead of using FILE_SUFFIX with a separate configuration file. It also updates the documentation to describe how to build the DFU-supported configuration.

### CODE REVIEW

**Review by Andrewpini — APPROVED** [review]

**Review by HaavardRei — APPROVED** [review]

File: samples/bluetooth/mesh/light/README.rst [hunk 1]
```diff
@@ -132,16 +132,17 @@ This sample is split into the following source files:
...
 
-     west build -b <BOARD> -p -- -DEXTRA_CONF_FILE="overlay-dfu.conf"
+     west build -b <BOARD> -p -- -DSB_CONF_FILE="sysbuild-dfu.conf" -DEXTRA_CONF_FILE="overlay-dfu.conf"
 
-The configuration overlay :file:`overlay-dfu.conf` enables the DFU feature.
-To review the required configuration alterations, open and inspect the :file:`overlay-dfu.conf` file.
+The configuration overlay :file:`overlay-dfu.conf` and the sysbuild configuration file :file:`sysbuild-dfu.conf` enable the DFU feature.
+To review the required configuration alterations, open and inspect the :file:`overlay-dfu.conf` file and the :file:`sysbuild-dfu.conf` file.
 For more information about using configuration overlay files, see :ref:`zephyr:important-build-vars` in the Zephyr documentation.
+For more information about selecting a sysbuild configuration file, see the section about the "Sysbuild Kconfig file" in the page :ref:`zephyr:sysbuild` in the Zephyr documentation.
```
Comment (mia-ko) [inline samples/bluetooth/mesh/light/README.rst:145]:
```suggestion
For more information about selecting a sysbuild configuration file, see the section about the sysbuild Kconfig file on the :ref:`zephyr:sysbuild` page in the Zephyr documentation.
```
Or alternatively: see the Sysbuild Kconfig file section on the XX page...

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by mia-ko — COMMENTED** [review]
Btw, does this require a changelog entry?

**Comment by ludvigsj** [comment]
> Btw, does this require a changelog entry?

I was thinking no, since this is a sample and won't affect existing users with existing applications? It's more of a consequence of the global switch to samples using sysbuild. But I can add one if you think it's needed?

**Comment by mia-ko** [comment]
> > Btw, does this require a changelog entry?
> 
> I was thinking no, since this is a sample and won't affect existing users with existing applications? It's more of a consequence of the global switch to samples using sysbuild. But I can add one if you think it's needed?

I guess this global switch will be mentioned somewhere, so we don’t have to do it sample by sample, so we drop the changelog entry.

**Review by mia-ko — APPROVED** [review]

**Pull request merged**